src/settings.js

```diff
@@ -214,6 +214,11 @@ var MEMORY_GROWTH_GEOMETRIC_CAP = 96*1024*1024;
 // and MEMORY_GROWTH_GEOMETRIC_CAP are ignored.
 var MEMORY_GROWTH_LINEAR_STEP = -1;
 
+// The "architecture" to compile for. 0 means the default wasm32, 1 is wasm64
+// for clang/lld but lowered to wasm32 in Binaryen, and 2 is the full end-to-end
```

Comment:
Do we want mode 1 in the long term? I thought it might have been just for interim testing until VMs catch up, but maybe I got that wrong.

If 1 is just temporary, then we may want a separate setting for it.

Reply:
I may be quite long until every engine supports `wasm64`, and maybe even then someone has a need to be compiling 64-bit code to an embedded `wasm32` engine.. so not intended to be very temporary.

That said, wether one or two settings doesn't matter by me.

Reply:
I see, thanks.

One setting sounds ok then. However, I think we should flip it: the "normal" case should be "1", so that users doing `-s SETTING` (which sets the value to 1) get what they expect. It seems like in the long term "1" should be the full end to end option, while "2" is something nonstandard (but useful).

Reply:
I'm fine with that too. The thinking behind the original numbers is one of "interpolation": 0 is no wasm64, 1 is halfway wasm64, and 2 is fully wasm64 :)